src/GHC/All.hs

```diff
@@ -84,8 +84,9 @@ data ParseError = ParseError
     }
 
 -- | Result of 'parseModuleEx', representing a parsed module.
-newtype ModuleEx = ModuleEx {
-    ghcModule :: Located HsModule
+data ModuleEx = ModuleEx {
+    ghcModule :: Located HsModule,
+    hlintExtensions :: [Extension]
```

Comment:
I think some name with `config` might be clearer - it took me a while to realise that this `hlint` means "it comes from invoking the `hlint` tool with certain CLI args or a configuration file"

```suggestion
    hlintConfigExtensions :: [Extension]
```
Perhaps?